omod/src/main/webapp/resources/js/view/list.js

```diff
@@ -0,0 +1,163 @@
+                        args[0] = 'item:' + event;
+                        args.splice(1, 0, self);
+                        // args = ["item:key:change", this=listEditor, itemEditor, fieldEditor]
+
+                        editors.InventoryList.prototype.trigger.apply(this, args);
+                    }, self);
+
+                    item.editor.on('change', function () {
+                        if (!item.addEventTriggered) {
+                            item.addEventTriggered = true;
+                            this.trigger('add', this, item.editor);
+                        }
+                        this.trigger('item:change', this, item.editor);
+                        this.trigger('change', this);
+                    }, self);
+
+                    item.editor.on('focus', function () {
+                        if (this.hasFocus) {
+                            return;
+                        }
+
+                        this.trigger('focus', this);
+                    }, self);
+                    item.editor.on('blur', function () {
+                        if (!this.hasFocus) {
+                            return;
+                        }
+
+                        var self = this;
+                        setTimeout(function () {
+                            if (_.find(self.items, function (item) {
+                                    return item.editor.hasFocus;
+                                })) {
+                                return;
+                            }
+
+                            self.trigger('blur', self);
+                        }, 0);
+                    }, self);
+
+                    if (userInitiated || value) {
+                        item.addEventTriggered = true;
+                    }
+
+                    if (userInitiated) {
+                        self.trigger('add', self, item.editor);
+                        self.trigger('change', self);
+                    }
+                };
+
+                //Check if we need to wait for the item to complete before adding to the list
+                if (this.Editor.isAsync) {
+                    item.editor.on('readyToAdd', _addItem, this);
+                } else {
+                    //Most editors can be added automatically
+                    _addItem();
+                }
+
+                return item;
+            }
+        });
+
+        /**
+         * A single item in the list
+         *
+         * @param {editors.List} options.list The List editor instance this item belongs to
+         * @param {Function} options.Editor   Editor constructor function
+         * @param {String} options.key        Model key
+         * @param {Mixed} options.value       Value
+         * @param {Object} options.schema     Field schema
+         */
+        editors.InventoryList.Item = editors.List.Item.extend({
+            events: {
+                'click [data-action="remove"]': function (event) {

```

Comment:
this data action is already defined in the backbonforms list.js. If you move that code there into a separate function, you could just override that function here and add your logic, rather than overriding the event... Meaning the delete operation still would be done by backboneforms but you just add your confirmation before that happens...